app/queries/occupation_standard_query.rb

```diff
@@ -15,14 +15,14 @@ def self.run(occupation_standards, search_term_params)
       .by_state_abbreviation(search_term_params[:state_abbreviation])
       .by_national_standard_type(search_term_params[:national_standard_type]&.keys)
       .by_ojt_type(search_term_params[:ojt_type]&.keys)
-      .and(
-        occupation_standards.by_title(search_term_params[:q])
-        .or(
-          occupation_standards.by_rapids_code(search_term_params[:q])
-        )
-        .or(
-          occupation_standards.by_onet_code(search_term_params[:q])
-        )
+      .where(
+        "title ILIKE :q OR rapids_code ILIKE :q OR onet_code ILIKE :q OR
+        occupation_standards.id IN (
+          SELECT occupation_standards.id FROM occupation_standards
+          JOIN industries ON occupation_standards.industry_id = industries.id
+          WHERE industries.name ILIKE :q
+        )",
+        q: "%#{search_term_params[:q]}%"
```

Comment:
@jenlu33 I think we could leave the existing code here as-is, and add another `or` segment for searching by industry name like:

```
.and(
  occupation_standards.by_title(search_term_params[:q])
  .or(
    occupation_standards.by_rapids_code(search_term_params[:q])
  )
  .or(
    occupation_standards.by_onet_code(search_term_params[:q])
  )
  .or(
    occupation_standards.by_industry_name(search_term_params[:q])
  )
)
```

And then the `occupation_standards.by_industry_name` method could join with industries and try to match on the name. What do you think about that? Happy to pair together.

Reply:
@littleforest I tried doing this before, but I would get this error:
`Relation passed to #or must be structurally compatible. Incompatible values: [:joins]`
When I looked it up, it seems like the issue is that the other scopes do not need a joins while the scope for `by_industry_name` does and it's causing problems when used with `or`

Reply:
We might be able to fake it by doing an isolated `join` query, and then returning a query that just grabs the ids. Like:

```
  scope :by_industry_name, ->(name) do
    if name.present?
      ids = OccupationStandard.joins(:industry).where("industries.name ILIKE ?", "%#{sanitize_sql_like(name).split.join("%")}%").pluck(:id)
      where(id: ids)   # <-- this will be the final scope that gets returned to be used with the "or" statement
    end
  end
```

Reply:
Thanks, I'll try this when I get the chance!